src/offlinelogstorage/dlt_offline_logstorage_behavior.c

```diff
@@ -53,17 +53,17 @@ unsigned int g_logstorage_cache_size;
  */
 void dlt_logstorage_log_file_name(char *log_file_name,
                                   DltLogStorageUserConfig *file_config,
-                                  char *name,
+                                  DltLogStorageFilterConfig *filter_config,
                                   int idx)
 {
-    if ((log_file_name == NULL) || (file_config == NULL))
+    if ((log_file_name == NULL) || (file_config == NULL) || (filter_config == NULL))
         return;
 
     char file_index[10] = { '\0' };
 
     /* create log file name */
     memset(log_file_name, 0, DLT_MOUNT_PATH_MAX * sizeof(char));
-    strcat(log_file_name, name);
+    strcat(log_file_name, filter_config->file_name);
```

Comment:
maybe strncat should be used here and in line 114 to make sure we don't overflow

Reply:
Agree. This required some refactoring of the entire function otherwise it wouldn't have served any point.